sui/gas-service.js

```diff
@@ -0,0 +1,220 @@
+        arguments: [
+            tx.object(gasServiceConfig.objects.GasService),
+            tx.object(gasServiceConfig.objects.GasCollectorCap),
+            tx.pure.address(receiver), // Receiver address
+            tx.pure.u64(atomicAmount), // Amount
+        ],
+    });
+
+    const receipt = await broadcast(client, keypair, tx);
+
+    printInfo('Gas collected', receipt.digest);
+}
+
+async function refund(keypair, client, gasServiceConfig, args, options) {
+    const walletAddress = keypair.toSuiAddress();
+
+    const gasServicePackageId = gasServiceConfig.address;
+    const gasServiceObjectId = gasServiceConfig.objects.GasService;
+
+    const [messageId, amount] = args;
+    const receiver = options.receiver || walletAddress;
+
+    const atomicAmount = getAtomicAmount(amount);
+
+    const bytes = await getBcsBytesByObjectId(client, gasServiceObjectId);
+    const { balance: gasServiceBalance } = gasServiceStruct.parse(bytes);
+
+    // Check if the gas service balance is sufficient
+    if (gasServiceBalance < atomicAmount) {
+        printError('Insufficient gas service balance', `${getFormattedAmount(gasServiceBalance)} < ${getFormattedAmount(atomicAmount)}`);
+        return;
+    }
+
+    const tx = new TransactionBlock();
+    tx.moveCall({
+        target: `${gasServicePackageId}::gas_service::refund`,
+        arguments: [
+            tx.object(gasServiceConfig.objects.GasService),
+            tx.object(gasServiceConfig.objects.GasCollectorCap),
+            tx.pure(bcs.string().serialize(messageId).toBytes()), // Message ID for the contract call
+            tx.pure.address(receiver), // Refund address
+            tx.pure.u64(atomicAmount), // Amount
+        ],
+    });
+
+    const receipt = await broadcast(client, keypair, tx);
+
+    printInfo('Gas refunded', receipt.digest);
+}
+
+async function processCommand(command, chain, args, options) {
+    const [keypair, client] = getWallet(chain, options);
+
+    await printWalletInfo(keypair, client, chain, options);
+
+    if (!chain.contracts.GasService) {
+        throw new Error('GasService contract not found');
+    }
+
+    await command(keypair, client, chain.contracts.GasService, args, options);
+}
+
+async function mainProcessor(options, args, processor, command) {
+    const config = loadSuiConfig(options.env);
+    await processor(command, config.sui, args, options);
+    saveConfig(config, options.env);
+}
+
+if (require.main === module) {
+    const program = new Command();
+
+    program.name('gas-service').description('Interact with the gas service contract.');
+
+    const payGasCmd = new Command()
+        .command('payGas <amount> <destinationChain> <destinationAddress> <channelId> <payload>')
+        .description('Pay gas for the new contract call.')
+        .option('--refundAddress <refundAddress>', 'Refund address. Default is the sender address.')
+        .option('--params <params>', 'Params. Default is empty.')
```

Comment:
is the default empty, or undefined, i forget? if it's undefined we should set it explicitly to empty bytes